### DISCUSSION TITLE
Trait input field validation

### DISCUSSION THREAD

**Discussion by confettidc** [GrapesJS/grapesjs#3945] [Q&A]
Hi,

I would like enhance the text input with "name" attribute value got some restriction.

I create my "input" type as below:
```
editor.DomComponents.addType('input', {
		// Make the editor understand when to bind `my-input-type`
		isComponent: el => el.tagName === 'INPUT',

		extend: 'input',
		model: {
			// Default properties
			defaults: {
				traits: [
					{
						type: 'text',
						label: 'Name',
						name: 'name',
						required: true
					},
					{
						type: 'select',
						label: 'Type',
						name: 'type',
						options: [
							{id: 'text', name: 'Text'},
							{id: 'number', name: 'Number'},
						]
					},
					'placeholder',
					{
						type: 'checkbox',
						label: 'Required',
						name: 'required'
					}
				]
			},
			// Model definition
			init() {
				this.on('change:attributes:name', this.handleNameChange);
			},
			handleNameChange() {
				var name = this.getAttributes().name;
		    	console.log('Input name changed to: ', name);

		    	if (name == 'name' || name == 'email' || name == 'mobile') {
		    		console.log("here");

		    		//editor.runCommand('core:undo');
		    		const component = editor.getSelected();
		    		component.getTrait('name').set('value', '');
		    		component.getTrait('name').props().value = '';

		    		editor.Modal.open({
						title: 'Error',
						content: name + ' fieldname is forbidden, please add as system field'
					});
		    	}
		    },
		},
		
	});

```

It successfully removed the "name" value in canvas and display the modal:
![圖片](https://user-images.githubusercontent.com/5524183/141739936-32b5c3db-3889-411e-bc46-0c9d2ab87c46.png)

![圖片](https://user-images.githubusercontent.com/5524183/141739984-ce369c1d-8744-4a22-aa9f-d74f248a373b.png)

However, the value still in trait settings:
![圖片](https://user-images.githubusercontent.com/5524183/141740048-9b9b3cb0-3132-4812-8e8c-70935d8909af.png)

How to remove the value in trait also?

Thanks.

**Comment by artf** [comment]
The trait reads always the data from the component so you should reupdate it in such a case `component.addAttributes({ name: '' })`

**Reply by confettidc** [reply]
Yes, this can make the value in canvas disappeared, but the typed value still exists in traits, any method to remove that value? Thanks.

![圖片](https://user-images.githubusercontent.com/5524183/142959829-b8aff1b6-24d5-465f-ae8d-6c54664db905.png)

**Reply by artf** [reply]
By checking the demo addAttributes triggers also the update of traits. Are you using the latest version?

**Reply by confettidc** [reply]
Yes, v0.17.29

And trait value only can update when reload page (editor)

Thanks.
